packages/components/src/text-control/stories/index.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta, ComponentStory } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import TextControl from '..';
+
+const meta: ComponentMeta< typeof TextControl > = {
+	component: TextControl,
+	title: 'Components/TextControl',
+	argTypes: {
+		onChange: {
+			action: 'onChange',
+			control: { type: null },
+		},
+		value: {
+			control: { type: null },
+		},
+	},
+	parameters: {
+		controls: {
+			expanded: true,
+		},
+		docs: { source: { state: 'open' } },
+	},
+};
+export default meta;
+
+const DefaultTemplate: ComponentStory< typeof TextControl > = ( {
+	...args
+} ) => {
+	const [ value, setValue ] = useState( '' );
+
+	return (
+		<TextControl
+			{ ...args }
+			value={ value }
+			onChange={ ( v ) => {
+				setValue( v );
+			} }
+		/>
+	);
+};
+
+export const Default: ComponentStory<
+	typeof TextControl
+> = DefaultTemplate.bind( {} );
+Default.args = {
+	label: 'Label Text',
+	hideLabelFromVision: false,
+	help: 'Help text to explain the input.',
+	type: 'text',
+};
```

Comment:
I think we'd prefer to have a default story as _vanilla_ as possible. We could add another story that showcases using `help` and `label`. There's no need to specify `hideLabelFromVision` and `type`, since those are already the default prop values.

```suggestion
export const Default: ComponentStory<
	typeof TextControl
> = DefaultTemplate.bind( {} );
Default.args = {};

export const WithLabelAndHelpText: ComponentStory<
	typeof TextControl
> = DefaultTemplate.bind( {} );
WithLabelAndHelpText.args = {
	...Default.args,
	label: 'Label Text',
	help: 'Help text to explain the input.',
};
```

Reply:
Sounds good to me.